samples: boards: st: adapt the mco to the nucleo_wba55

FRASTM opened the pull request:
The stm32wba55 nucleo board has the PB8 output pin for the MCO, it cannot be used for the console RX
signal. This PR is routing the PB8 to the MCO instead of the usart1 during this sample execution

To allow the    /samples/boards/st/mco execution on the nucleo_wba55cg, just remove the RX pin of the console
--> this is the MCO output 
The console usart1 is kept functional on it TX pin, the usart1 Rx pin is not really used for this application

erwango commented on samples/boards/st/mco/boards/nucleo_wba55cg.overlay:
```diff
@@ -13,12 +13,14 @@
 	chosen {
 		zephyr,bt-c2h-uart = &lpuart1;
 		zephyr,uart-pipe = &lpuart1;
```
Not related but these 2 lines could actually be removed

FRASTM replied:
removed

mathieuchopstm commented on samples/boards/st/mco/boards/nucleo_f411re.overlay:
```diff
@@ -1,5 +1,11 @@
 /* Enable the PLLI2s and set it as clock source for the MCO2 (with prescaler 5) */
 
+#define MCO_PRE_DIV_2 4
```
This should come from `include/zephyr/dt-bindings/clock/stm32f4_clock.h`.
Is the include missing, or is this dead code?

djiatsaf-st replied:
May be related to this https://github.com/zephyrproject-rtos/zephyr/pull/81726#discussion_r1957919852 ?

FRASTM replied:
yes, this is already defined

mathieuchopstm commented on samples/boards/st/mco/boards/nucleo_f411re.overlay:
```diff
@@ -1,5 +1,11 @@
 /* Enable the PLLI2s and set it as clock source for the MCO2 (with prescaler 5) */
 
+#define MCO_PRE_DIV_2 4
+
+&clk_hsi {
+	status = "okay";
+};
```
Nit: We could use a comment indicating what uses HSI.
```suggestion
/* Enable HSI as it will be output on MCO pin by this sample */
&clk_hsi {
	status = "okay";
};
```

FRASTM replied:
adding comment

FRASTM committed:
samples: boards: st: build only this sample for stm32 boards

Makes the sample build only in automatic test CI
Running on the nucleo_f411re board requires hsi enable
Running on the nucleo_wba55cg can still output the console
on the usart1 Tx even if the usart1Rx is used by the MCO output

Signed-off-by: Francois Ramu <francois.ramu@st.com>

mathieuchopstm commented on samples/boards/st/mco/boards/nucleo_wba55cg.overlay:
```diff
@@ -8,17 +8,19 @@
 	status = "okay";
 };
-/* MCO use same pin like usart1 RX - disable usart */
+
+/* MCO uses same pin like usart1 RX - disable usart RX pin */
+
```
Maybe this comment is no longer needed since we have it in `&usart1 {}` now?

FRASTM commented:
rebase on 3d2059f9b66faf2b6cf4be15b2f4adc2f092f42b
decision to build only that samples\boards\st\mco\ (no way to automaticallly test the MCO signal)
 Running on the nucleo_f411re board requires hsi enabling
    Running on the nucleo_wba55cg can still output the console
    on the usart1 Tx even if the usart1Rx is used by the MCO output

mathieuchopstm reviewed (approved):
Nit: Changes should be split across 3 commits. Otherwise LGTM.
